lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp

```diff
@@ -534,12 +591,22 @@ void SingleExitLoopTransformer::ensureSingleExitBlock() {
         if (blocksToBeMoved.count(succ) > 0 || succ == nearestCommonPD) {
           continue;
         }
+        // Clone `succ` if it is reachable by a node outside of this loop.
         if (!DI->properlyDominates(header, succ)) {
-          // Split this edge so that we don't mess up arguments passed in
-          // from other predecessors of succ.
-          if (succ->getNumArguments() > 0) {
-            splitEdge(current->getTerminator(), edgeIdx, DI, LI);
+          // Backedge of a parent loop? This can happen when we have labeled
+          // breaks in loops. We cannot clone the blocks in this case.
+          if (DI->properlyDominates(succ, header)) {
```

Comment:
in the example, i believe header is block 4, and no `succ` dominates header -- is this if body exercised somewhere?

Reply:
Yes, the sumOfProductsWithBound function with labeled breaks in our tests exercises this code: 
https://github.com/apple/swift/blob/835215fe8fb06db8d36ed747e0b36f3cfc951cb9/test/TensorFlowRuntime/sese_loop_canonicalization.swift#L69

Reply:
nice!

The comment says "We cannot clone the blocks in this case.", but the body does `cloneBlock()`. Can you clarify?